node/src/ledger/ledger_manager.rs

```diff
@@ -0,0 +1,515 @@
+use mina_p2p_messages::v2::{
+    LedgerHash, MinaBaseAccountBinableArgStableV2, MinaBaseSparseLedgerBaseStableV2,
+    MinaLedgerSyncLedgerAnswerStableV2, MinaLedgerSyncLedgerQueryStableV1,
+    MinaStateProtocolStateValueStableV2, NonZeroCurvePoint, StateHash,
+};
+use openmina_core::block::ArcBlockWithHash;
+use openmina_core::channels::{mpsc, oneshot};
+use openmina_core::snark::{Snark, SnarkJobId};
+use std::collections::{BTreeMap, BTreeSet};
+use std::sync::{Arc, Mutex};
+
+use super::ledger_service::LedgerCtx;
+use crate::account::AccountPublicKey;
+use crate::block_producer::vrf_evaluator::DelegatorTable;
+use crate::block_producer::{BlockProducerWonSlot, StagedLedgerDiffCreateOutput};
+use crate::ledger::LedgerAddress;
+use crate::p2p::channels::rpc::StagedLedgerAuxAndPendingCoinbases;
+use crate::rpc::RpcScanStateSummaryScanStateJob;
+use crate::transition_frontier::sync::{
+    ledger::staged::StagedLedgerAuxAndPendingCoinbasesValid,
+    TransitionFrontierRootSnarkedLedgerUpdates,
+};
+use crate::transition_frontier::CommitResult;
+
+/// The type enumerating different requests that can be made to the
+/// service. Each specific constructor has a specific response
+/// constructor associated with it. Unfortunately, this relationship
+/// can't be expressed in the Rust type system at the moment. For this
+/// reason this type is private while functions wrapping the whole call
+/// to the service are exposed as the service's methods.
+enum LedgerRequest {
+    AccountsSet {
+        snarked_ledger_hash: LedgerHash,
+        parent: Arc<Mutex<LedgerAddress>>,
+        accounts: Vec<MinaBaseAccountBinableArgStableV2>,
+    }, // expected response: LedgerHash
+    BlockApply {
+        block: ArcBlockWithHash,
+        pred_block: ArcBlockWithHash,
+    }, // expected response: Success
+    ChildHashesGet {
+        snarked_ledger_hash: LedgerHash,
+        parent: LedgerAddress,
+    }, // expected response: ChildHashes
+    Commit {
+        ledgers_to_keep: BTreeSet<LedgerHash>,
+        root_snarked_ledger_updates: TransitionFrontierRootSnarkedLedgerUpdates,
+        needed_protocol_states: BTreeMap<StateHash, MinaStateProtocolStateValueStableV2>,
+        new_root: Arc<Mutex<ArcBlockWithHash>>,
```

Comment:
A general comment about all these `Arc<Mutex<...>>` values. Given that the original values are not already wrapped in `Arc<Mutex<...>>` you will have to clone their anyway, so I don't see any advantage in this extra wrapping here.